File: src/main/java/com/epicsquid/whoosh/init/WhooshMenuTypes.java
package com.epicsquid.whoosh.init;

import com.epicsquid.whoosh.Whoosh;
import com.epicsquid.whoosh.containers.TransporterMenu;
import com.epicsquid.whoosh.gui.TransportScreen;
import com.tterrag.registrate.Registrate;
import com.tterrag.registrate.util.entry.MenuEntry;

public class WhooshMenuTypes {

	private static final Registrate REGISTRATE = Whoosh.registrate();

	public static final MenuEntry<TransporterMenu> TRANSPORTER = REGISTRATE.menu("transporter", TransporterMenu::new, () -> TransportScreen::new).register();

	public static void init() {
	}
}

File: src/main/java/com/epicsquid/whoosh/registery/WhooshMenuTypes.kt
package com.epicsquid.whoosh.registery

import com.epicsquid.whoosh.containers.TransporterMenu
import com.epicsquid.whoosh.gui.TransportScreen
import com.tterrag.registrate.Registrate
import com.tterrag.registrate.builders.MenuBuilder.ScreenFactory
import dev.epicsquid.whoosh.Whoosh
import net.minecraft.network.FriendlyByteBuf
import net.minecraft.network.chat.Component
import net.minecraft.world.entity.player.Inventory
import net.minecraft.world.inventory.MenuType

object WhooshMenuTypes {
    private val REGISTRATE: Registrate = Whoosh.registrate
    @JvmField
	val TRANSPORTER = REGISTRATE.menu<TransporterMenu, TransportScreen>(
        "transporter",
        { type: MenuType<TransporterMenu?>?, id: Int, inventory: Inventory?, buf: FriendlyByteBuf? ->
            TransporterMenu(
                type,
                id,
                inventory,
                buf
            )
        }) {
        ScreenFactory { container: TransporterMenu?, inv: Inventory?, component: Component? ->
            TransportScreen(
                container,
                inv,
                component
            )
        }
    }
        .register()

    fun init() {}
}
